hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -109,6 +109,9 @@ public HoodieFileGroupReader(HoodieReaderContext<T> readerContext,
     readerContext.setLatestCommitTime(latestCommitTime);
     readerContext.setShouldMergeUseRecordPosition(shouldUseRecordPosition);
     readerContext.setHasLogFiles(!this.logFiles.isEmpty());
+    if (readerContext.getHasLogFiles() && start != 0) {
+      throw new IllegalArgumentException("Filegroup reader is doing log file merge but not reading from the start of the base file");
```

Comment:
Is it easy to add tests on these two validation?

Reply:
For the log file one it is easy. For bootstrap it is hard

Reply:
https://issues.apache.org/jira/browse/HUDI-8169

Reply:
Added bootstrap validation but it will never hit until that issue ticket above is completed.